From 59f564c5f81f53cb80b1bb3c2e263495bba2ce94 Mon Sep 17 00:00:00 2001
From: Ilkka Koskinen <ilkka.koskinen@intel.com>
Date: Thu, 18 Sep 2014 18:32:56 -0700
Subject: [PATCH 13/28] atomisp2: Fix pm_qos handling in probe()'s failure path

The driver tried to remove pm_qos settings twice

Change-Id: I2fd33e7b8ae62045ce311de1a3ac8b1f50c536a1
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9279
Original-Tracked-On: https://jira01.devtools.intel.com/browse/OAM-7550
Signed-off-by: Ilkka Koskinen <ilkka.koskinen@intel.com>
Signed-off-by: Per Persson <per.persson@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/83121
Build: Dessert M COHO (SC)
Build: Dessert M BXT RVP (SC)
Approver: Yunliang Ding <yunliang.ding@intel.com>
Reviewed-by: Yunliang Ding <yunliang.ding@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/440499
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c    | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
index 732cf2a..2988b96 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
@@ -1582,9 +1582,6 @@ enable_msi_fail:
 fw_validation_fail:
 	release_firmware(isp->firmware);
 load_fw_fail:
-#ifdef CONFIG_GMIN_INTEL_MID
-	pm_qos_remove_request(&isp->pm_qos);
-#endif
 	pci_dev_put(isp->pci_root);
 	return err;
 }
-- 
1.9.1

